### PR TITLE
Extract analysis library

### PR DESCRIPTION
**Pull request by dstallenberg**

### CODE REVIEW

File: libraries/analysis/lib/factories/SourceFactory.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if (!parsed.base) {
+      throw new Error(`File '${filePath}' is not a file`);
+    }
+
+    if (!parsed.ext) {
+      throw new Error(`File '${filePath}' does not have an extension`);
+    }
+
+    if (!existsSync(filePath)) {
+      throw new Error(`File ${filePath} does not exist`);
```
Comment (mitchellolsthoorn) [inline libraries/analysis/lib/factories/SourceFactory.ts:34]: These errors should go in the diagnostics file

File: libraries/analysis/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@syntest/analysis",
+  "version": "0.0.0",
+  "description": "A analysis package",
```
Comment (mitchellolsthoorn) [inline libraries/analysis/package.json:4]:
```suggestion
  "description": "An analysis package",
```

**Review by mitchellolsthoorn — CHANGES_REQUESTED** [review]
You might want to cherrypick the last two commits onto a clean branch. Since this one has a ton of merge conflitcs

File: libraries/analysis/package.json (hunk 2 above)
Comment (dstallenberg) [reply libraries/analysis/package.json:4]: Fixed.

File: libraries/analysis/lib/factories/SourceFactory.ts (hunk 1 above)
Comment (dstallenberg) [reply libraries/analysis/lib/factories/SourceFactory.ts:34]: Fixed.

File: libraries/analysis/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
+## 0.2.0 (2023-02-17)
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply libraries/analysis/CHANGELOG.md:6]: Fixed.

**Review by mitchellolsthoorn — APPROVED** [review]

**Pull request merged**